### PR TITLE
add dev tests

### PR DESCRIPTION
**Pull request by Intron7**

### CODE REVIEW

File: .github/workflows/test-gpu-dev.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+          cache-dependency-path: pyproject.toml
+
+      - name: Install rapids-singlecell
+        run: uv pip install --system -e .[test,rapids12] --extra-index-url=https://pypi.nvidia.com
+
+      - name: Install scanpy-dev
+        run: uv pip install --system git+https://github.com/scverse/scanpy.git
+
+      - name: Install anndata-dev
+        run: uv pip install --system git+https://github.com/scverse/anndata.git
```
Comment (flying-sheep) [inline .github/workflows/test-gpu-dev.yml:59]: once more: never do multiple `[uv] pip install` calls in sequence.

File: .github/workflows/test-gpu-dev.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+
+      - name: Pip list
+        run: pip list
```
Comment (flying-sheep) [inline .github/workflows/test-gpu-dev.yml:63]: I don’t think this is really necessary, `uv` tells you what it installed, right?

File: .github/workflows/test-gpu-dev.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
+name: GPU-CI-DEV
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    types:
+      - labeled
+      - opened
+      - synchronize
```
Comment (flying-sheep) [inline .github/workflows/test-gpu-dev.yml:10]: maybe also `reopened`? but maybe that’s implicitly done.

**Review by flying-sheep — CHANGES_REQUESTED** [review]
looks generally good, but please remember the thing I keep saying about sequential `pip installs`

File: .github/workflows/test-gpu-dev.yml (hunk 1 above)
Comment (Intron7) [reply .github/workflows/test-gpu-dev.yml:59]: ok

File: .github/workflows/test-gpu-dev.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      - name: Install UV
+        uses: hynek/setup-cached-uv@v2
+        with:
+          cache-dependency-path: pyproject.toml
+
+      - name: Install rapids-singlecell
+        run: >-
+          uv pip install --system -e .[test,rapids12]
+          git+https://github.com/scverse/scanpy.git
+          git+https://github.com/scverse/anndata.git
```
Comment (flying-sheep) [inline .github/workflows/test-gpu-dev.yml:56]:
```suggestion
          "scanpy @ git+https://github.com/scverse/scanpy.git"
          "anndata @ git+https://github.com/scverse/anndata.git"
```

**Review by flying-sheep — APPROVED** [review]

**Pull request merged**